tagger.go

```diff
@@ -65,85 +69,240 @@ func newLinodeClient() linodego.Client {
 	return client
 }
 
-func checkLinodeTagsAgainstConfig(linodes []linodego.Instance, rules []TagRule) (instanceTagMap, error) {
-	linodeIDTagMap := make(instanceTagMap)
-
-	for _, linode := range linodes {
-		tags := linode.Tags
-		sort.Strings(tags)
-		var combinedNewTags []string
-		for _, rule := range rules {
-			validInstance := regexp.MustCompile(rule.Regex)
-
-			if validInstance.MatchString(linode.Label) {
-				var newTags []string
-
-				// check `absent` tags to remove unwanted tags
-				for _, tag := range tags {
-					if !slices.Contains(rule.Tags.Absent, tag) {
-						// if this tag is not on the `absent` list,
-						// we can persist it through to the new tag set
-						newTags = append(newTags, tag)
-					}
+func compareTags(linodeObject interface{}, linodeTags []TagRule) (combinedNewTags []string, tags []string, linodeID int) {
```

Comment:
The named return values read a bit odd here with the given names (especially when we explicitly provide the return values when we call anyway). Personally, I'd rather drop the named return values and just add a comment indicating that the function returns, in order, a slice of new tags, a slice of the old tags, and the object id.

Reply:
```suggestion
func compareTags(linodeObject interface{}, tagRules []TagRule) (combinedNewTags []string, tags []string, linodeID int) {
```
Similarly here too, these aren't necessarily just linode compute instances, and I think this can benefit from an updated name to reflect that